### ISSUE TITLE
Add `workspace.package` and `workspace.dependencies` to avoid duplicate

### ISSUE THREAD

**Issue by silathdiir** [privacy-scaling-explorations/zkevm-circuits#1650]
### Describe the feature you would like

Reference [cargo-workspace-deduplicate](https://rust-lang.github.io/rfcs/2906-cargo-workspace-deduplicate.html), [workspace-package](https://doc.rust-lang.org/nightly/cargo/reference/workspaces.html#the-package-table) and [workspace-dependecies](https://doc.rust-lang.org/nightly/cargo/reference/workspaces.html#the-dependencies-table), suppose if we could add `workspace.package` and `workspace.dependencies` to root `Cargo.toml`:
```
[workspace.package]
version = "0.1.0"
edition = "2021"
authors = ["The appliedzkp team"]
license = "MIT OR Apache-2.0"

[workspace.dependencies]
env_logger = "0.9"
log = "0.4"
rand = "0.8"
...
```
Then we could update `Cargo.toml` in sub-crates:
```
[package]
name = "zkevm-circuits"
version.workspace = true
edition.workspace = true
authors.workspace = true
license.workspace = true

[dependencies]
env_logger.workspace = true
log.workspace = true
rand.workspace = true
...
```

[cargo-dependency-inheritor](https://github.com/TimonPost/cargo-dependency-inheritor) could be used to fix automatically (but seems we may need some manual check for different crate versions and format).

**Comment by ChihChengLiang** [comment]
I like the idea and I think we should do it. We can have more consistency in dependencies between crates.
@ed255 or @KimiWu123, can you provide a second opinion on this?

**Comment by KimiWu123** [comment]
Good idea! However, it usually occurs some troubles to manage dependencies among sub-moduels like this proposal, but it's nice to give it a try and see how it goes.

**Comment by ed255** [comment]
Sounds good to me to try this change!  It can clean up the dependency management, specially for the `halo2_proofs` which has a hardcoded git url and tag in many of the subcrates.
